osgp/platform/osgp-adapter-domain-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/domain/da/infra/jms/core/messageprocessors/GetDataResponseMessageProcessor.java

```diff
@@ -42,65 +40,31 @@
             @Qualifier("domainDistributionAutomationInboundOsgpCoreResponsesMessageProcessorMap") final MessageProcessorMap messageProcessorMap) {
-        super(responseMessageSender, messageProcessorMap, MessageType.GET_DATA);
+        super(responseMessageSender, messageProcessorMap, GET_DATA);
     }
 
     @Override
-    public void processMessage(final ObjectMessage message) throws JMSException {
+    public void processMessage(final ObjectMessage message) {
         LOGGER.debug("Processing DA GET_DATA response message");
         this.getResponseValues(message).ifPresent(this::processResponseValues);
     }
 
-    private Optional<ResponseValues> getResponseValues(final ObjectMessage message) {
-        final ResponseValues responseValues = new ResponseValues();
+    private Optional<ResponseMessage> getResponseValues(final ObjectMessage message) {
         try {
-            responseValues.correlationUid = message.getJMSCorrelationID();
-            responseValues.messageType = message.getJMSType();
-            responseValues.organisationIdentification = message
-                    .getStringProperty(Constants.ORGANISATION_IDENTIFICATION);
-            responseValues.deviceIdentification = message.getStringProperty(Constants.DEVICE_IDENTIFICATION);
-            responseValues.responseMessage = (ResponseMessage) message.getObject();
-            responseValues.responseMessageResultType = responseValues.responseMessage.getResult();
-            responseValues.osgpException = responseValues.responseMessage.getOsgpException();
-            responseValues.payload = (String) responseValues.responseMessage.getDataObject();
+            return Optional.of((ResponseMessage) message.getObject());
         } catch (final JMSException e) {
-            this.logJmsException(responseValues, e);
+            LOGGER.error("UNRECOVERABLE ERROR, unable to read ObjectMessage instance, giving up.", e);
             return Optional.empty();
         }
-        return Optional.of(responseValues);
     }
 
-    private void processResponseValues(final ResponseValues responseValues) {
+    private void processResponseValues(final ResponseMessage response) {
         try {
-            LOGGER.info("Calling application service function to handle response: {}", responseValues.messageType);
-            this.adHocManagementService.handleGetDataResponse(responseValues.payload,
-                    responseValues.deviceIdentification);
-        } catch (final Exception e) {
-            this.handleError(e, responseValues.correlationUid, responseValues.organisationIdentification,
-                    responseValues.deviceIdentification, responseValues.messageType);
+            this.adHocManagementService.handleGetDataResponse(response);
+        } catch (final RuntimeException e) {
+            this.handleError(e, response.getCorrelationUid(), response.getOrganisationIdentification(),
+                    response.getDeviceIdentification(), GET_DATA.toString());
```

Comment:
response.getMessageType() is null in this case. I used the hardcoded value, because it's the only message type handled by the class. We could improve this, but I'll wait for the architectural discussion on how we want to pass payload and metadata through the components.